### DISCUSSION TITLE
Drag-and-Drop automation through UI5 actions

### DISCUSSION THREAD

**Discussion by ph-alsvik** [ui5-community/wdi5#211] [Ideas]
Hi JS-Soft!

I see on your roadmap that you are planning to implement drag-and-drop testing in WDI5. The company I work for is working on an app with drag-and-drop functionality which we hope to automate the testing of, since its a key feature of the app. By using actions from ```sap.ui.test.actions``` library, I’ve made some simple additions to the WDI5 codebase to make this work like this:

In ```injectUI5.js```:

```javascript
window.wdi5.simulateDragDrop = (oSourceControl, oDestinationControl) => {
                        const oDrag = new Drag()
                        oDrag.executeOn(oSourceControl)
                        const oDrop = new Drop({
                            aggregationName: "items"
                        })
                        oDrop.executeOn(oDestinationControl)
                    }
```

Then execute drag-and-drop inside test script through the ```executeAsync``` function.

I was wondering if you think this is a good approach to getting drag-and-drop testing to work, or if you think there is a better approach than using actions from the UI5 library? 

One immediate issue I can see is that this would require version 1.76 or higher, but not sure how big of issue this actually is.

**Comment by vobu** [comment]
that sounds absolutely reasonable and a good approach!
might also be worth extending the function signature to pass in the `aggregation` name (as optional parameter):
```js
window.wdi5.simulateDragDrop = (oSourceControl, oDestinationControl, sAggregationName = "items") => {
  const oDrag = new Drag()
  oDrag.executeOn(oSourceControl)
  const oDrop = new Drop({
      aggregationName: sAggregationName
  })
  oDrop.executeOn(oDestinationControl)
}
```
if you don't mind, whoop it into a PR and we'll get it in - if the need arises to extend that functionality, we can always iterate over your starter.

**Reply by ph-alsvik** [reply]
I'm on it! 😄
